DBZ-9556 Introduce SwapGeometryCoordinates transformation (#6827)
debezium/debezium: documentation/modules/ROOT/pages/transformations/swap-geometry-coordinates.adoc

[id="swap-geometry-coordinates"]
= SwapGeometryCoordinates

:toc:
:toc-placement: macro
:linkattrs:
:icons: font
:source-highlighter: highlight.js

toc::[]

> 

Reviewer: ```suggestion
In environments where you must send data from a source database that uses one coordinate system to a target that uses a different system, you can use the `SwapGeometryCoordinates` SMT  to reformat the data.
```

When serializing geometry information into a WKB (Well-Known Binary) or EWKB (Extended Well-Known Binary) byte array, the source database will choose to use one of two different coordinate systems:

Legacy CRS:84::
This is an older convention that serializes coordinates using `(x, y)` or `(longitude, lattitude)` axis order.

European Petroleum Survey Group (EPSG)::

> This is a modern convention that serializes coordinates using `(y, x)` or `(lattitude, longitude)` axis order.

Reviewer: ```suggestion
A modern convention that serializes coordinates by using `(y, x)` or `(latitude, longitude)` axis order.
```

The format irrelevant in a {prodname} source connector since the data is written to the change event using the source database's chosen coordinate system.

> However, for a consumer, it's essential to understand the coordinate system used by the source and what is expected at the target.

Reviewer: ```suggestion
However, a sink connector must not only be able to interpret the coordinate system of the source database, but also provide data to the target system in  a format that it can consume.
```

> If the target expects a different coordinate system than the source, this SMT can be used to swap between the coordinate systems.

Reviewer: ```suggestion

```
Move to beginning of the topic.

[[example-swap-geometry-coordinates]]
== Example

> To swap between the two coordinate systems, configure the `SwapGeometryCoordinates` SMT in the Kafka Connect configuration for a connector.

Reviewer: ```suggestion
To convert the coordinate system of data that a {prodname} connector captures from a source database to a different coordinate system, add the `SwapGeometryCoordinates` SMT to the connector configuration.
```

> You can configure the transformation to apply to the default set of spatial reference identifiers or modify the identifiers used to tailor which geometry fields are swapped and which are left unchanged.

Reviewer: ```suggestion
You can configure the transformation to transform the default set of spatial reference identifiers, or customize the configuration to apply it only to specific geometry fields.
Other fields are left unchanged.
```

For example, to swap between coordinate systems using the default set of spatial reference identifiers, add the following lines to your connector configuration:

[source]
----
transforms=swap
transforms.swap.type=io.debezium.transforms.SwapGeometryCoordinates
----

> To limit the swap of coordinate systems to only spatial reference identifier `4326`, add the following lines to your connector configuration:

Reviewer: ```suggestion
To apply the transformation only to spatial reference identifier `4326`, add the following lines to your connector configuration:
```

[source]
----
transforms=swap
transforms.swap.type=io.debezium.transforms.SwapGeometryCoordinates
transforms.swap.srids=4326
----

[[swap-geometry-coordinates-configuration-options]]
== Configuration options

The following table lists the configuration options that you can use with the `SwapGeometryCoordinates` SMT.

.SwapGeometryCoordinates SMT configuration options
[cols="14%a,40%a,10%a, 16%a, 16%a, 10%a"]
|===
|Property
|Description
|Type
|Default
|Valid Values
|Importance

|[[swap-geometry-coordinates-srids]]<<swap-geometry-coordinates-srids, `srids`>>

> |A comma-separated list of spatial reference identifiers used to control what geometry values will have their coordinate systems swapped.

Author: Looks good to me.

Other fields are left unchanged.
For example, to configure the connector to convert the default set of spatial reference identifiers, add the following lines to the connector configuration:

[source]
----
transforms=swap
transforms.swap.type=io.debezium.transforms.SwapGeometryCoordinates
----

To apply the transformation only to spatial reference identifier `4326`, add the following lines to your connector configuration:

[source]
----
transforms=swap
transforms.swap.type=io.debezium.transforms.SwapGeometryCoordinates
transforms.swap.srids=4326
----

[[swap-geometry-coordinates-configuration-options]]
== Configuration options

The following table lists the configuration options that you can use with the `SwapGeometryCoordinates` SMT.

.SwapGeometryCoordinates SMT configuration options
[cols="14%a,40%a,10%a, 16%a, 16%a, 10%a"]
|===
|Property
|Description
|Type
|Default
|Valid Values
|Importance

|[[swap-geometry-coordinates-srids]]<<swap-geometry-coordinates-srids, `srids`>>
|A comma-separated list of spatial reference identifier values to which you want to apply the transformation.

> The transformation swaps the coordinate system of the specified values.

Reviewer: ```suggestion
The transformation swaps the coordinate system only for the specified values.
```
